lib/puppet/network/http/connection.rb

```diff
@@ -197,6 +197,8 @@ def request_with_redirects(request, options)
     end
 
     def apply_options_to(request, options)
```

Comment:
Good question. For puppet (agent, webrick, passenger), `Puppet::Network::HttpPool.http_instance(url.host, url.port, use_ssl)` will return an instance of the default http client class `Puppet::Network::HTTP::Connection` and that's the only case where we would try to set the `User-Agent` above (in the call to Connection#apply_options_to).

Puppetserver on the other hand overrides the [http client class](https://github.com/puppetlabs/puppetserver/blob/fa34cedb5ba38f86ab156b40c007fe25fae600b1/src/ruby/puppetserver-lib/puppet/server/config.rb#L29), but in that case, it implements its own [post](https://github.com/puppetlabs/puppetserver/blob/master/src/ruby/puppetserver-lib/puppet/server/http_client.rb#L53) (and `get`) logic. That said, I don't know if puppetserver sets `User-Agent` for connections it makes, e.g. to puppetdb, and whether we want puppetserver to read the puppet settings for that purpose? /cc @cprice404 @camlow325